vendor/k8s.io/kubernetes/pkg/controller/statefulset/stateful_set.go

```diff
@@ -450,7 +450,11 @@ func (ssc *StatefulSetController) sync(key string) error {
 // syncStatefulSet syncs a tuple of (statefulset, []*v1.Pod).
 func (ssc *StatefulSetController) syncStatefulSet(set *apps.StatefulSet, pods []*v1.Pod) error {
 	glog.V(4).Infof("Syncing StatefulSet %v/%v with %d pods", set.Namespace, set.Name, len(pods))
-	if err := ssc.control.UpdateStatefulSet(set, pods); err != nil {
+	setCopy, err := api.Scheme.DeepCopy(set)
```

Comment:
> this should probably go deeper into UpdateStatefulSet when we know the mutation will happen, will tweak this if this confirms the tests pass.

It should go deeper, but if this works it's ok to bring out the big hammer to make progress as long as you still chase it down.